api/src/main/java/org/apache/iceberg/expressions/CountStar.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.expressions;
+
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.StructLike;
+
+public class CountStar<T> extends CountAggregate<T> {
+  protected CountStar(BoundTerm<T> term) {
+    super(Operation.COUNT_STAR, term);
+  }
+
+  @Override
+  protected Long countFor(StructLike row) {
+    return 1L;
+  }
+
+  @Override
+  protected Long countFor(DataFile file) {
+    long count = file.recordCount();
+    if (count < 0) {
+      return null;
+    }
```

Comment:
Curious when would this ever be negative? or is this just for this logic to be defensive against bad metadata?

Reply:
Some imported Avro files had incorrect metadata several versions ago. I don't think it is widespread, but it is good to handle it.